sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -0,0 +1,1507 @@
+              needsLitWarning = needsLitWarning,
+              note = note)
+          }
+        case _ if note.isDefined || needsLitWarning =>
+          new PartiallySupported(needsAsterisks, needsLitWarning = needsLitWarning, note = note)
+        case _ =>
+          new Supported(needsAsterisks)
+      }
+    }
+  }
+}
+
+object TypeSig {
+  /**
+   * Create a TypeSig that only supports a literal of the given type.
+   */
+  def lit(dataType: TypeEnum.Value): TypeSig =
+    TypeSig.none.withLit(dataType)
+
+  /**
+   * Create a TypeSig that has partial support for the given type.
+   */
+  def psNote(dataType: TypeEnum.Value, note: String): TypeSig =
+    TypeSig.none.withPsNote(dataType, note)
+
+  /**
+   * All types nested and not nested
+   */
+  val all: TypeSig = new TypeSig(TypeEnum.values, TypeEnum.values)
+  /**
+   * No types supported at all
+   */
+  val none: TypeSig = new TypeSig(TypeEnum.ValueSet())
+
+  val BOOLEAN: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.BOOLEAN))
+  val BYTE: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.BYTE))
+  val SHORT: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.SHORT))
+  val INT: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.INT))
+  val LONG: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.LONG))
+  val FLOAT: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.FLOAT))
+  val DOUBLE: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.DOUBLE))
+  val DATE: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.DATE))
+  val TIMESTAMP: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.TIMESTAMP))
+  val STRING: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.STRING))
+  val DECIMAL: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.DECIMAL))
+  val NULL: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.NULL))
+  val BINARY: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.BINARY))
+  val CALENDAR: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.CALENDAR))
+  /**
+   * ARRAY type support, but not very useful on its own because no nested types under
+   * it are supported
+   */
+  val ARRAY: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.ARRAY))
+  /**
+   * MAP type support, but not very useful on its own because no nested types under
+   * it are supported
+   */
+  val MAP: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.MAP))
+  /**
+   * STRUCT type support, but only matches empty structs unless you add nested types to it.
+   */
+  val STRUCT: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.STRUCT))
+  /**
+   * User Defined Type (We don't support these in the plugin yet)
+   */
+  val UDT: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.UDT))
+
+  /**
+   * A signature for types that are generally supported by the plugin. Please make sure to check
+   * what Spark actually supports instead of blindly using this in a signature.
+   */
+  val legacySupportedTypes: TypeSig = BOOLEAN + BYTE + SHORT + INT + LONG + FLOAT + DOUBLE + DATE +
```

Comment:
legacy makes me think this is old deprecated types (it might just be me though), would it be better to say something like basic types or commonSupportedTypes. Not a big deal though as I can't think of anything great either.